### DISCUSSION TITLE
mod_oauth2 and mod_auth_openidc in same web server

### DISCUSSION THREAD

**Discussion by mshaddock** [OpenIDC/mod_oauth2#56] [Q&A]
Is it possible to use both mod_oauth2 and mod_auth_openidc in the same web server?  We are using mod_auth_openidc, but we have a requirement that some applications need to be able to use curl against this server, and it looks like recommendation for that is to use mod_oauth2.

**Comment by zandbelt** [comment]
yes, they've been designed and tested to do so

**Reply by studersi** [reply]
Does this mean that they can also be used for the same location? For example, the same path on the same Virtual Host is accessed once with an OIDC session cookie and once with an OAuth2 Access Token?

**Reply by zandbelt** [reply]
yes, indeed, both modules "listen" to `AuthType auth-openidc`

**Reply by studersi** [reply]
Interesting, thanks for the clarification.
